examples/example-6/example-6.cpp

```diff
@@ -1,3 +1,36 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright 2017-2019
+
+Karlsruhe Institute of Technology
+Universitat Jaume I
+University of Tennessee
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
```

Comment:
I guess it automatically got added due to the file being changed to .cpp.

Reply:
Yes, the question is if we should remove it or not. To prevent it, we could add this file as an exception in the `add_license.sh` script, or rename it to `*.cc` again.
Personally, I think we should remove it, but I don't know enough about licensing to actually know what the best thing to do is here.

Reply:
If you keep it as `.cpp` as you did, this license will get added back again.

Reply:
By removing it, I meant doing either add it as an exception in the `add_license.sh` or simply rename it back to `.cc` (and probably add `.cc` to the list of accepted endings in doxygen).
I would prefer the ladder, but I am (like I said) not 100% sure if it is necessary.

Reply:
I think it is fine if both licenses are there in the raw file. In the documentation, there is a note saying that it is modified to add a ginkgo solver. So, I'll leave it as .cpp for uniformity.

Reply:
OK, like I said, I have too little knowledge about licensing to know what is the right thing to do.
I guess since you modified it, it makes sense that our license is there as well as theirs.

I will mark it as resolved then.